ml-algorithms/src/main/java/org/opensearch/ml/engine/ingest/AbstractIngestion.java

```diff
@@ -75,6 +75,55 @@ protected double calculateSuccessRate(List<Double> successRates) {
             );
     }
 
+    /**
+     * Filters fields in the map where the value contains the specified source index as a prefix.
+     * When there is only one source file, users can skip the source[] prefix
```

Comment:
If there is only one source file, is it ok for cx to input `source[0].xxx` now ?

Reply:
Yes it works for both cases. this function makes sure it finds the fields with source[0] or any fields that does not prefixed with "source".